sunpy/instr/aia/response.py

```diff
@@ -0,0 +1,308 @@
+"""
+Calculate SDO/AIA wavelength and temperature response functions.
+"""
+
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+import os
+import warnings
+from collections import namedtuple
+
+import numpy as np
+import h5py
+import astropy.units as u
+import astropy.constants as const
+from scipy.interpolate import splev, splrep
+try:
+    from ChiantiPy.tools.data import MasterList
+except ImportError:
+    warnings.warn('Cannot import ChiantiPy. You will not be able to calculate the temperature response functions.')
+
+from sunpy.util.config import get_and_create_download_dir
+from sunpy.util.net import check_download_file
+from .response_utils import aia_instr_properties_to_table, make_emiss_table, EmissTableInterface
+
+SSW_AIA_REMOTE_PATH = 'https://hesperia.gsfc.nasa.gov/ssw/sdo/aia/response/'
+AIA_INSTR_FILES = ['aia_V6_all_fullinst.genx', 'aia_V6_fuv_fullinst.genx']
+
+__author__ = ["Tessa D. Wilkinson", "Will Barnes"]
+
+
+class Response(object):
+    """
+    Calculate the SDO/AIA wavelength and temperature response functions.
+
+    This class calculates the wavelength and temperature response functions for
+    the channels of the Atmospheric Imaging Assembly (AIA) instrument on
+    the Solar Dynamics Observatory spacecraft. The response functions are
+    calculated using AIA instrument information obtained from
+    Solar Software (SSW) .genx files. Temperature response functions are
+    calculated using atomic data from the CHIANTI atomic database. See [1]_ for
+    more details.
+
+    Parameters
+    ----------
+    channel_list : `list`, optional
+    instrument_files : `list`, optional
+        If no instrument files are supplied and they do not exist locally, the
+        necessary files are automatically downloaded.
+
+    Examples
+    --------
+    >>> from sunpy.instr.aia import Response
+    >>> response = Response()
+    >>> wavelength_response = response.calculate_wavelength_response()
+    >>> temperature_response = response.calculate_temperature_response()
+
+    Notes
+    -----
+    Calculating the temperature response functions requires tabulated emissivities,
+    for both the continuum and line contributions, for each ion. The first time the temperature
+    response functions are calculated, this table is automatically generated. You can also
+    generate your own emissivity table using `make_emiss_table` and then passing the appropriate
+    filename to the temperature response function method.
+
+    References
+    ----------
+    .. [1] Boerner et al., 2012, Sol. Phys., `275, 41
+        <http://adsabs.harvard.edu/abs/2012SoPh..275...41B>`_
+    """
+
+    def __init__(self, channel_list=[94, 131, 171, 193, 335, 211, 304], instrument_files=None):
```

Comment:
lists should not be defined in function definition blocks,
```python 
def __init__(self, channel_list=None):
    if channel_list is None:
        channel_list = [94, 131, 171, 193, 335, 211, 304]
```
is safer, because the default list can be changed after function definition.